lib/instream/query/url.ex

```diff
@@ -118,7 +118,13 @@ defmodule Instream.Query.URL do
         false -> "?"
       end
 
-    "#{url}#{glue}#{key}=#{URI.encode(value)}"
+    encoded_value =
+      case String.contains?(url, "&") do
+        true -> "#{URI.encode_www_form(value)}"
+        false -> "#{URI.encode(value)}"
+      end
+
+    "#{url}#{glue}#{key}=#{encoded_value}"
```

Comment:
Would it be sufficient to just always use `URI.encode_www_form(value)`? There shouldn't be any case that relies on that previously broken behaviour, right?

Reply:
there is one test case which is failing, hence thought to have specific condition

Reply:
I think the change might be `URI.encode/1` using `+` to encode a space and `URI.encode_www_form/1` using `%20`. But looking at the InfluxDB documentation always using `curl --url-encode` and therefore percent encoding I think the test is at fault and not the fix.

Will look at some more official clients (like the official CLI client) how they handle the encoding and then adjust this fix accordingly.

Reply:
Can you take a look at https://github.com/mneudert/instream/commit/ae92a6a98a87358fabcf76e4559db8d4c9d6d090 to see if that solves your problem?

From what I tested the real solution would be to use `URI.encode_query/1` instead of the other variants. That constructs the most likely expected encoding for spaces as `%20` is usually only used for path encoding while `+` is used for parameter value encoding. I modified a test case to ensure both question marks and ampersands are encoded.

If that works for you I would update your PR and merge (and asap release a new version including the fix). I just did not want to mess around on your `master` branch without any notice :)

Reply:
Perfect!!!
It worked